File: java/eyes.images.java/src/main/java/com/applitools/eyes/images/ImagesCheckSettings.java
package com.applitools.eyes.images;

import com.applitools.eyes.Region;
import com.applitools.eyes.fluent.CheckSettings;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.awt.image.BufferedImage;

@JsonIgnoreProperties({
        "image",
        "path",
})
public class ImagesCheckSettings extends CheckSettings implements IImagesCheckTarget {

    private BufferedImage image;
    private String path;

    public ImagesCheckSettings(BufferedImage image){
        this.image = image;
    }

    public ImagesCheckSettings(String path) {
        this.path = path;
    }

    public ImagesCheckSettings(BufferedImage image, Region region){
        this.image = image;
        updateTargetRegion(region);
    }

    public BufferedImage getImage() {
        return image;
    }

    public String getPath() {
        return path;
    }

    @Override
    public ImagesCheckSettings clone(){
        ImagesCheckSettings clone = new ImagesCheckSettings(this.image);
        super.populateClone(clone);
        clone.image = this.image;
        clone.path = this.path;
        return clone;
    }

    @Override
    public ImagesCheckSettings densityMetrics(int xDpi, int yDpi) {
        return (ImagesCheckSettings) super.densityMetrics(xDpi, yDpi);
    }

    @Override
    public ImagesCheckSettings densityMetrics(int xDpi, int yDpi, Double scaleRatio) {
        return (ImagesCheckSettings) super.densityMetrics(xDpi, yDpi, scaleRatio);
    }
}

File: dotnet/Eyes.Images.DotNet/Images/ImagesCheckSettings.cs
using System;
using System.Drawing;
using Applitools.Utils;

namespace Applitools.Images
{
    public class ImagesCheckSettings : CheckSettings, IImagesCheckTarget
    {
        private Bitmap image_;
        private Uri imageUri_;
        private string base64_;

        public ImagesCheckSettings(Bitmap image)
        {
            image_ = image;
        }

        public ImagesCheckSettings(string base64)
        {
            base64_ = base64;
        }

        public ImagesCheckSettings(Uri imageUri)
        {
            imageUri_ = imageUri;
        }

        private ImagesCheckSettings() { }

        Bitmap IImagesCheckTarget.Image => image_;

        Uri IImagesCheckTarget.ImageUri => imageUri_;

        public ImagesCheckSettings Region(Rectangle region)
        {
            ImagesCheckSettings clone = Clone_();
            clone.UpdateTargetRegion(region);
            return clone;
        }

        public string ToImage()
        {
            if (image_ != null)
            {
                return image_.ToBase64();
            }

            if (imageUri_ != null)
            {
                return imageUri_.ToString();
            }

            return base64_;
        }

        private ImagesCheckSettings Clone_()
        {
            return (ImagesCheckSettings)Clone();
        }

        protected override CheckSettings Clone()
        {
            ImagesCheckSettings clone = new ImagesCheckSettings();
            base.PopulateClone_(clone);
            clone.image_ = this.image_;
            clone.imageUri_ = this.imageUri_;
            clone.base64_ = this.base64_;
            return clone;
        }
    }
}
